doc/Administration-guide/System-installation/Cluster-setup/index.md

```diff
@@ -5,62 +5,121 @@ permalink: /:path/
+
+In this guide we assume that you had already set up your SQL and NoSQL database clusters, so this tutorial doesn't cover such themes like setting up ones for Cassandra, MongoDB or PostgreSQL. 
+Refer to official [Cassandra](http://docs.datastax.com/en/landing_page/doc/landing_page/current.html), [MongoDB](https://docs.mongodb.com/manual/) and [PostgreSQL](https://www.postgresql.org/docs/) documentation in order to setup corresponding database cluster.
 
-This page describes how to setup and configure kaa-node cluster.
+MariaDB come with out-of-the-box master-master replication support thus we recommend to use this database in your Kaa cluster. 
+You can find detailed instructions in [MariaDB cluster setup guide]({{root_url}}Administration-guide/System-installation/Cluster-setup/MariaDB-cluster-setup-guide/).
 
-In general cluster setup is similar to [single node setup](../Single-node-installation), except few details. We need at least 3 nodes to create a reliable cluster. For simplicity in this tutorial we will be using single instance of SQL and NoSQL databases, so this tutorial doesn't cover such themes like setting up Cassandra, MongoDB or PostgreSQL cluster setup. Let us assume that we have 3 nodes with Ubuntu 14.04 installed on each of them.
+> Note: In a cluster you need to connect to databases from external hosts so you need to allow such external connections in corresponding database security configurations and configure firewall rules for database host machine. 
+> Refer to official documentation for corresponding database for security configuration details.
+
+In addition to added on ```kaa-node``` service installation step firewall rules on every endpoint Kaa Administrator need to add some more rules for databases (depending on a set databases in the cluster) and Zookeeper ports.
+
+> Note: This configuration will not affect AWS deployment or any other cloud provider and must be applied only if you setting up cluster on VMs or separate Linux machines. 
+> Next instructions depend on specific cloud provider. 
 
 ```bash
-node1 172.1.1.1
-node2 172.2.2.2 
-node3 172.3.3.3
+# MongoDB port
+$ sudo iptables -I INPUT -p tcp -m tcp --dport 27017 -j ACCEPT
+$ sudo iptables -I OUTPUT -p tcp -m tcp --dport 27017 -j ACCEPT
+
+# Cassandra port
+$ sudo iptables -I INPUT -p tcp -m tcp --dport 9042 -j ACCEPT
+$ sudo iptables -I OUTPUT -p tcp -m tcp --dport 9042 -j ACCEPT
+
+# MariaDB port 
+$ sudo iptables -I INPUT -p tcp -m tcp --dport 3306 -j ACCEPT
+$ sudo iptables -I OUTPUT -p tcp -m tcp --dport 3306 -j ACCEPT
+
+# PostgreSQL port
+$ sudo iptables -I INPUT -p tcp -m tcp --dport 5432 -j ACCEPT
+$ sudo iptables -I OUTPUT -p tcp -m tcp --dport 5432 -j ACCEPT
+
+# Zookeeper port
+$ sudo iptables -I INPUT -p tcp -m tcp --dport 2181 -j ACCEPT
```

Comment:
7000 port is also required, but i don't recall who is using it. close all ports and check who is failing